components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -205,6 +206,14 @@ func (s *Principal) GetTemplateSecrets(ctx context.Context, client sdk.Client, o
 	templateSecrets["ImageRegistryPassword"] = ecrCredentials.Password
 	templateSecrets["ImageRegistryEndpoint"] = ecrCredentials.Endpoint
 
+	assumedRoleCredentials, err := client.GetRoleCredentials(roleArn, env.PrincipalCredentialsSessionDuration()).Get()
```

Comment:
not overlay loving having a env.Blah() call burried in here ... ideally the only place configuring stuff from environment would be around the main/controller setup.